### PR TITLE
Add `import_to_main` flag for Main module package name conflicts resolution

### PR DESCRIPTION
**Pull request by krynju**
This contains only the `import_to_main` part of https://github.com/JuliaLang/PackageCompiler.jl/pull/794

Fixes https://github.com/JuliaLang/PackageCompiler.jl/issues/768

Changelog:
1. added `import_into_main` flag (default true), which allows you to control whether the top level packages will be imported into Main in the sysimage
2. added skip & warnings on the import to Main name conflict https://github.com/JuliaLang/PackageCompiler.jl/issues/768

### CODE REVIEW

File: src/PackageCompiler.jl [hunk 1]
```diff
@@ -409,10 +410,21 @@ function create_sysimg_object_file(object_file::String,
...
 
     # Make packages available in Main. It is unclear if this is the right thing to do.
-    for pkg in packages
-        print(julia_code_buffer, """
-            import $pkg
+    if import_into_main
+        for pkg in packages
+            print(julia_code_buffer, """
+                if isdefined(Main, Symbol("$pkg"))
+                    @warn(
```
Comment (KristofferC) [inline src/PackageCompiler.jl:417]: Did you check that users will see this warning?

Comment (krynju) [reply src/PackageCompiler.jl:417]: I saw it on a build once but let me confirm on a simple example

**Comment by krynju** [comment]
@KristofferC Can we merge and tag this? I confirmed it works fine on my case

File: src/PackageCompiler.jl [hunk 2]
```diff
@@ -504,6 +516,9 @@ compiler (can also include extra arguments to the compiler, like `-g`).
    transitive dependencies into the sysimage. This only makes a difference if some
    packages do not load all their dependencies when themselves are loaded. Defaults to `true`.
 
+- `import_into_main::Bool`: If `true`, import all packages from the sysimage into `Main`.
```
Comment (KristofferC) [inline src/PackageCompiler.jl:519]: Is this true? I thought it was just the packages in the `[deps]` section of the project file.

Comment (krynju) [reply src/PackageCompiler.jl:519]: No, I believe only deps from the project you're creating a sysimage of go in 
```suggestion
- `import_into_main::Bool`: If `true`, import all packages from `packages` into `Main`.
```

This is more accurate I guess

**Pull request merged**